lib/ui/painting/flutter_rtree.h

```diff
@@ -0,0 +1,103 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_LIB_UI_PAINTING_FLUTTER_RTREE_H_
+#define FLUTTER_LIB_UI_PAINTING_FLUTTER_RTREE_H_
+
+#include "third_party/skia/include/core/SkBBHFactory.h"
+#include "third_party/skia/include/core/SkTypes.h"
+
+namespace flutter {
+/**
+ * An R-Tree implementation based on
+ * https://chromium.googlesource.com/skia/+/master/src/core/SkRTree.h. In short,
```

Comment:
This link will go stale fast. Hit `y` on the GitHub link to get a permalink to the implementation you used as basis for this. So something like https://github.com/google/skia/blob/93d75eff08b069a87e78a25156f249ab95bc7c95/src/core/SkRTree.h maybe.